engine/core/IgeTextureMap.ts

```diff
@@ -593,14 +603,14 @@ export class IgeTextureMap extends IgeTileMap2d {
 	 * @return {*}
 	 * @private
 	 */
-	_renderTile (ctx, x, y, tileData, tileEntity, rect, sectionX, sectionY) {
+	_renderTile (ctx, x, y, tileData, tileEntity, rect: IgeRect | undefined = undefined, sectionX = undefined, sectionY = undefined) {
```

Comment:
To fit in with the rest of the refactor, these can probably be written as `_renderTile (ctx, x, y, tileData, tileEntity, rect?: IgeRect, sectionX?: number, sectionY?: number) {`